runtime/oti/j9accessbarrier.h

```diff
@@ -113,11 +113,59 @@ typedef struct J9IndexableObject* mm_j9array_t;
 		? (&(((elemType*)J9_CONVERT_POINTER_FROM_TOKEN_VM__(javaVM, ((U_32*)(((UDATA)(array)) + (javaVM)->discontiguousIndexableHeaderSize))[((U_32)index)/((javaVM)->arrayletLeafSize/sizeof(elemType))]))[((U_32)index)%((javaVM)->arrayletLeafSize/sizeof(elemType))])) \
 		: (&(((elemType*)(((UDATA*)(((UDATA)(array)) + (javaVM)->discontiguousIndexableHeaderSize))[((U_32)index)/((javaVM)->arrayletLeafSize/sizeof(elemType))]))[((U_32)index)%((javaVM)->arrayletLeafSize/sizeof(elemType))])))
 
+#define J9JAVAARRAYCONTIGUOUS_WITHOUT_DATAADDRESS_EA(vmThread, array, index, elemType) \
+	(J9VMTHREAD_COMPRESS_OBJECT_REFERENCES(vmThread) \
+		? (&((elemType*)((((J9IndexableObjectContiguousCompressed *)(array)) + 1)))[index]) \
+		: (&((elemType*)((((J9IndexableObjectContiguousFull *)(array)) + 1)))[index]))
+
+#define J9JAVAARRAYCONTIGUOUS_WITHOUT_DATAADDRESS_EA_VM(javaVM, array, index, elemType) \
```

Comment:
I don't like the fact that there is a reference to DATAADDRESS in the name outside of appropriate build flag
and there are 2 issues:
- DATAADDRESS is outside of J9VM_ENV_DATA64
- the build flag should really be GC_SPARSE_HEAP_ALLOCATION (not J9VM_ENV_DATA64)

I'll think if we can reshuffle cleanly

Reply:
perhpas the simplest is to use _BASE_ (or some other generic term) rather than _WITHOUT_DATAADDRESS_
@dmitripivkine, opinion?

for using the proper build flag, as we discussed recently, let's don't change it yet https://github.com/eclipse-openj9/openj9/pull/20503#discussion_r1829653850, we can do it

Reply:
yes, we can use BASE